app/src/main/java/fr/free/nrw/commons/theme/BaseActivity.java

```diff
@@ -44,4 +47,17 @@ protected void onDestroy() {
         super.onDestroy();
         compositeDisposable.clear();
     }
+
+    /**
+     * its function is apply fontScale on device
+     */
+    public  void adjustFontScale( Configuration configuration,float scale) {
+        configuration.fontScale = scale;
+        DisplayMetrics metrics = getResources().getDisplayMetrics();
```

Comment:
metrics can have final modifier.

Reply:
@Prince-kushwaha Would you mind trying `final DisplayMetrics metrics` (and same for `wm`)?

Discussion of the merits of using final local variables: https://stackoverflow.com/questions/316352/why-would-one-mark-local-variables-and-method-parameters-as-final-in-java